docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+                };
+                maturity_modulation_disabled = opt true:bool;
+                max_number_of_principals_per_neuron = opt 42:nat64;
+            };
+    }
+)'
+```
+
+## `AddGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type AddGenericNervousSystemFunction : NervousSystemFunction;
+
+    type NervousSystemFunction = record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt FunctionType;
+    };
+
+    type FunctionType = variant {
+        NativeNervousSystemFunction : record {};
+        GenericNervousSystemFunction : GenericNervousSystemFunction;
+    };
+
+    type GenericNervousSystemFunction = record {
+        validator_canister_id : opt principal;
+        target_canister_id : opt principal;
+        validator_method_name : opt text;
+        target_method_name : opt text;
+    };
+```
+
+### Putting it together
+
+```candid
+    type AddGenericNervousSystemFunction: record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt variant {
+            NativeNervousSystemFunction : record {};
+            GenericNervousSystemFunction : record {
+                validator_canister_id : opt principal;
+                target_canister_id : opt principal;
+                validator_method_name : opt text;
+                target_method_name : opt text;
+            }
+        };
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            AddGenericNervousSystemFunction = record {
+                id = nat64;
+                name = text;
+                description = opt text;
+                function_type : opt variant {
+                    NativeNervousSystemFunction = record {};
+                    GenericNervousSystemFunction = record {
+                        validator_canister_id = opt principal "ltyfs-qiaab-aaaak-aan3a-cai";
+                        target_canister_id = opt principal;
+                        validator_method_name = opt text;
+                        target_method_name = opt text;
+                    }
+                };
+            }
+        };
+    }
+)'
+```
+
+## `RemoveGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type RemoveGenericNervousSystemFunction : nat64;
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            RemoveGenericNervousSystemFunction = 123456789:nat64;
+        };
+    }
+)'
+```
+
+## `UpgradeSnsToNextVersion`
+
```

Comment:
```suggestion
Recall that all approved SNS canister versions are stored on the NNS canister [SNS-W](introduction/sns-architecture). 
New SNS canister wasm codes are approved by NNS proposals and then added to SNS-W. 
Each SNS community can then simply decide if and when they want to upgrade their SNS instance to the next SNS version that is available on SNS-W. 
To do so, they can use an `UpgradeSnsToNextVersion`proposal. 
If this proposal is adopted, it will trigger a call to SNS root that will ask SNS-W which new wasm version is available and then upgrade to this new version. 
```

I think eventually it would be good to define this in more detail, but better to have already something now!